File: ecommerce-ui/src/app/services/product.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { API_URLS } from '../api.urls';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  constructor(private http:HttpClient) { }

  addProduct(data:any){
    return this.http.post(API_URLS.BASE_URL+API_URLS.SELLER_PRODUCT,data);
  }

  getProducts(pageNumber:number,pageSize:number){
    return this.http.get(`${API_URLS.BASE_URL}${API_URLS.PUBLIC_LISTING_API}?pageNumber=${pageNumber}&pageSize=${pageSize}`)
  }

  deleteProduct(productId:string){
    return this.http.delete(`${API_URLS.BASE_URL}${API_URLS.SELLER_PRODUCT}/${productId}`);
  }

  updateProduct(productId:string,data:any){
    return this.http.put(`${API_URLS.BASE_URL}${API_URLS.SELLER_PRODUCT}/${productId}`,data);
  }

  searchProducts(query:string){
    return this.http.get(`${API_URLS.BASE_URL}${API_URLS.PUBLIC_SEARCH_API}?query=${query}`)
  }
}

File: Ecommerce-backend/src/main/java/com/project/service/ProductService.java
package com.project.service;

import com.project.dto.PaginatedResponse;
import com.project.dto.ProductDto;
import com.project.entity.User;

import java.util.List;

public interface ProductService {

    ProductDto addProduct(ProductDto productDTO, User user);

    PaginatedResponse<ProductDto> getProducts(int pageNumber, int pageSize, User user);

    String deleteProduct(String productId, User user);

    ProductDto updateProduct(String productId, ProductDto productDTO, User user);

    List<ProductDto> searchProductsByName(String query);
}
